src/main/java/backend/RepoIO.java

```diff
@@ -51,14 +52,32 @@ public RepoIO(boolean isTestMode, boolean enableTestJSON) {
 
     public CompletableFuture<Model> openRepository(String repoId) {
         if (repoStore.isRepoStored(repoId)) {
-            return repoStore.loadRepository(repoId)
-                .thenCompose(this::updateModel)
-                .exceptionally(withResult(new Model(repoId)));
+            return loadRepositoryFromStore(repoId);
         } else {
-            return repoSource.downloadRepository(repoId)
+            return downloadRepositoryFromSource(repoId);
+        }
+    }
+
+    private CompletableFuture<Model> loadRepositoryFromStore(String repoId) {
+        return repoStore.loadRepository(repoId)
+                .thenCompose(this::updateModel)
+                .exceptionally(ex -> {
+                    try {
```

Comment:
`exceptionally`'s mechanism is different from a normal `try/catch`. It's a proper part of the stream's monadic chain and only deal with exceptions which has been wrapped using `CompletableFuture.completeExceptionally` by 1 of the `CompletableFuture` earlier in the chain.

As such, `exceptionally` handles a special control flow path in which it requires a unwrapped return value for which I use `CompletableFuture.get` to obtain, in order to preserve the same stream programming style. From the *[javadocs](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get--)*, `CompletableFuture.get` will throw various exceptions (in the traditional way) in various scenarios.

To summarize, the code won't compile if the `try/catch` is removed.